lms/djangoapps/verify_student/views.py

```diff
@@ -1158,7 +1158,17 @@ def results_callback(request):
         'platform_name': settings.PLATFORM_NAME,
     }
     if result == "PASS":
-        log.debug(u"Approving verification for %s", receipt_id)
+        # If this verification is not an outdated version then make expiry date of previous approved verification NULL
+        # Setting expiry date to NULL is important so that it does not get filtered in the management command
+        # that sends email when verification expires : verify_student/send_verification_expiry_email
+        if attempt.status != 'approved':
+            verification = SoftwareSecurePhotoVerification.objects.filter(status='approved', user_id=attempt.user_id)
+            if verification:
+                log.info(u'Making expiry date of previous approved verification NULL for {}'.format(attempt.user_id))
+                previous_verification = verification.latest('updated_at')
+                SoftwareSecurePhotoVerification.objects.filter(pk=previous_verification.pk
```

Comment:
There is nothing wrong I can see from how this is implement. However, I suspect because we filter by pk, and then do update, this may meant we needed to do 2 round trips to database for this single line of code. Not that important consider the number of times we get into case. 
That said, I am curious why you felt we can't just do
```
previous_verfication.expiry_date = None
previous_verfication.expiry_email_date = None
previous_verification.save()
```

Reply:
The `updated_at` field in sspv model has `auto_now` set to True, which means any time I **save()** the model instance, updated_at will change. Some of the existing functionality of verification (showing your verification has expired on dashboard) relies on updated_at. I did not want to cause any inconsistencies that's why opted to **update** instead of save

Reply:
@zainab-amir Interesting. Thank you for the explanation. I wish the above explanation is also commented in the code. Do you mind just create a comment PR to help the next poor soul stumbles into this area of the code?

Reply:
yes sure.